src/cargo/util/toml/mod.rs

```diff
@@ -1737,12 +1738,22 @@ impl TomlManifest {
             debug!("manifest has no build targets");
         }
 
-        if let Err(e) = unique_build_targets(&targets, package_root) {
-            warnings.push(format!(
-                "file found to be present in multiple \
-                 build targets: {}",
-                e
-            ));
+        if let Err(conflict_targets) = unique_build_targets(&targets, package_root) {
+            conflict_targets.iter().for_each(|c| {
+                warnings.push(format!(
+                    "file `{}` found to be present in multiple \
+                 build targets:\n{}",
+                    c.0.display().to_string(),
+                    c.1.iter()
+                        .map(|t| format!(
+                            "  * This file exist in `{}` as a `{}` target.",
```

Comment:
Just a suggestion on the wording, I would maybe avoid repeating the same sentence multiple times. I think something like:

```
* `bin` target `foo`
* `bin` target `bar`
```

would be sufficient to get the point across. I wish it were easier to get line numbers from `Cargo.toml` so it could point directly where it is defined, but alas that isn't supported.

Reply:
Good suggestion! Thanks!